### PR TITLE
feat(CN-114): added feature for saving email/sms drafts and dropdown fields changed to multiselect

### PR DESCRIPTION
**Pull request by INX1**
Deploy:
API
client
Translation (I added English only for now because there is a lot of new ones, when it is deployed we should add for other languages)
changes to DB, take a look at the file
[CN-114.txt](https://github.com/koja5/custom_management/files/9823294/CN-114.txt)

### CODE REVIEW

File: server/routes/api.js [hunk 1]
```diff
@@ -9601,4 +9601,322 @@ router.get('/getClinicEmployees/:id', function (req, res) {
...
+  connection.getConnection(function (err, conn) {
+    if (err) {
+      logger.log("error", err.sql + ". " + err.sqlMessage);
+      res.json(err);
+    }
+
+    response = null;
+    logger.log("req.body: ", req.body)
+    var data = {
+      birthdayFrom: req.body.birthdayFrom,
```
Comment (koja5) [inline server/routes/api.js:9639]: We have here an identical object with the same property.
Why do we additional mapped property?
For example, if we need to add a new property in this way we need to change it in three places (front, back, and database).
Why we didn't create SQL query and send a directly req.body object?
**insert email_drafts set ? where id = '" + req.body.id + "'", req,body**
This also applies to other methods

File: src/app/component/dashboard/marketing/massive-email/massive-email.component.scss [hunk 2]
```diff
@@ -0,0 +1,51 @@
+.button-padding > button {
```
Comment (koja5) [inline src/app/component/dashboard/marketing/massive-email/massive-email.component.scss:1]: This looks like the same code, but repeated two times(for email and SMS). Why?
Why we don't make this one global SCSS file and use one code in different places...

File: src/app/component/dashboard/marketing/massive-email/massive-email.component.ts [hunk 3]
```diff
@@ -37,10 +50,63 @@ export class MassiveEmailComponent implements OnInit {
       .getConfiguration("administarator", "massive-email")
       .subscribe((config) => {
         this.configField = config;
+        this.getEmailDrafts();
+
         this.loading = false;
       });
   }
 
+  getEmailDrafts(selectNewlyCreated?: boolean) {
```
Comment (koja5) [inline src/app/component/dashboard/marketing/massive-email/massive-email.component.ts:59]: All these new functions also repeat two times(for email and SMS). It looks pretty the same.
It's problem for future development and maintenance. If we need to change something or add, we need to do it on a few places.
If the same code repeats a few times it does not need to be copy-paste into different components.
It's needed to make reusable components (or services).

**Review by koja5 — CHANGES_REQUESTED** [review]
Please check the comments. Some parts need to change.
Also, you have a change in the database, but you didn't give me SQL queries for it.
Check again our **step-by-step deploy document** on Jira.
This is an example of how @Lazarevicka describes hew pull request:

Deploy:
API
client
Translation
changes to DB, take a look at the file
[CN-13.txt](https://github.com/koja5/custom_management/files/9710372/CN-13.txt)

**Comment by INX1** [comment]
I did some changes to make it more reusable now, but some things need to be the same because of local states of components. In order to not have some copies, we would need to refactor this components into one and that would take a lot of time and is not in scope of this task

**Review by koja5 — APPROVED** [review]

**Review by koja5 — APPROVED** [review]

**Pull request merged**